### PR TITLE
feat(app-platform): Add sentry app details endpoint

### PR DESCRIPTION
**Pull request by MeredithAnya**
* Adds sentry app details endpoint
* Only superusers should have access to both published and unpublished apps (for now)

Refs API-172

### CODE REVIEW

File: src/sentry/api/endpoints/sentry_app_details.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            return Response(status=404)
+
+        try:
+            sentry_app = SentryApp.objects.get(id=sentry_app_id)
+        except SentryApp.DoesNotExist:
+            return Response(status=404)
+
+        # Superusers have access to the app, published or unpublished. Other
+        # users only have access to a published app (for now)
+        if request.user.is_superuser or sentry_app.status == SentryAppStatus.PUBLISHED:
```
Comment (markstory) [inline src/sentry/api/endpoints/sentry_app_details.py]: Should this logic live in a permission class? The `has_object_permissions` and `has_permission` hooks would map well to the permissions logic you have for list and details end points so far.

File: tests/sentry/api/endpoints/test_sentry_app_details.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        assert response.status_code == 200
+        assert response.data == {
+            'name': self.unpublished_app.name,
+            'scopes': [],
+            'uuid': self.unpublished_app.uuid,
+            'webhook_url': self.unpublished_app.webhook_url,
+        }
+
+    def test_users_only_see_published_apps(self):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_sentry_app_details.py]: How do normal users see their unpublished app?

Comment (mnoble) [reply tests/sentry/api/endpoints/test_sentry_app_details.py]: They don't, yet. "Unpublished" is effectively "development mode" where only those within the Org who owns it, can see it. The logic to enable the "only those within the Org" part of that will come in a followup PR. We're just trying to get the simple, happy-path, version of these endpoints in place first.

File: src/sentry/api/endpoints/sentry_app_details.py (hunk 1 above)
Comment (mnoble) [reply src/sentry/api/endpoints/sentry_app_details.py]: Hmm, yea, probably. Some of this is going to come down to the role you have within an Org too, which also probably makes sense to be in the permissions class. 👍

File: src/sentry/api/urls.py [hunk 3]
```diff
@@ -1091,6 +1092,11 @@
         SentryAppsEndpoint.as_view(),
         name='sentry-api-0-sentry-apps'
     ),
+    url(
+        r'^sentry-apps/(?P<sentry_app_slug>[^\/]+)/$',
```
Comment (adhiraj) [inline src/sentry/api/urls.py]: can we add `sentry-apps` to RESERVED_ORGANIZATION_SLUGS

File: src/sentry/api/bases/sentryapps.py [hunk 4]
```diff
@@ -11,16 +9,16 @@
 
 class SentryAppDetailsPermission(ScopedPermission):
     def has_object_permission(self, request, view, sentry_app):
-        return sentry_app.owner in request.user.get_orgs()
+        return request.user.is_superuser or sentry_app.owner in request.user.get_orgs()
```
[the remark this answers is not in the document]
Comment (MeredithAnya) [reply src/sentry/api/bases/sentryapps.py]: That should have changed in https://github.com/getsentry/sentry/pull/10107

**Review by markstory — APPROVED** [review]
Looks reasonable to me. I assume this will continue to evolve as sentry-apps develop.

**Pull request merged**